experanto/interpolators.py

```diff
@@ -478,6 +479,64 @@ def rescale_frame(self, frame: np.array) -> np.array:
         )
 
 
+class TimeIntervalInterpolator(Interpolator):
+    def __init__(self, root_folder: str, cache_data: bool = False, **kwargs):
+        super().__init__(root_folder)
+        self.cache_data = cache_data
+
+        meta = self.load_meta()
+        self.labels = list(sorted(meta["labels"].keys()))
+        self.start_time = meta["start_time"]
+        self.end_time = meta["end_time"]
+        self.valid_interval = TimeInterval(self.start_time, self.end_time)
+
+        if self.cache_data:
+            self.label_data = {
+                label: np.load(self.root_folder / f"{label}.npy", allow_pickle=True)
+                for label in self.labels
+            }
+
+    def interpolate(self, times: np.ndarray) -> tuple[np.ndarray, np.ndarray]:
+        """Interpolates over TimeInterval modality.
+
+        The key idea is the following: we have a meta.yml file whose "main" key is called labels. The keys of meta["labels"] are strings containing the labels of the time intervals we are interested.
+
+        Examples of labels for tiers are: train, validation and test.
+        Other examples of labels: saccade, gaze, target.
+
+        The values of meta["labels"] are filenames to .npy files and the .npy files contain np.array's of shape (n, 2), where n is the number of keys we have, say, for training, and 2 is the starting and the end time of the respective key.
+
+        The interpolation works as follows: the output signal array, out, is a np.array of shape (len(valid_times), number of labels). The values are boolean and we compute them as follows: out[valid_time, label] = 1 if start <= valid_times[i] <= end for one of the (start, end) pairs in the respective label.npy file, and 0 otherwise.
+        """
+        valid = self.valid_times(times)
+        valid_times = times[valid]
+
+        n_labels = len(self.labels)
+        n_times = len(valid_times)
+
+        if n_times == 0:
+            warnings.warn(
+                "TimeIntervalInterpolator returns an empty array, no valid times queried."
+            )
+            return np.empty((0, n_labels), dtype=bool), valid
+
+        out = np.zeros((n_times, n_labels), dtype=bool)
+
+        for l, label in enumerate(self.labels):
+            if self.cache_data:
+                label_intervals = self.label_data[label]
+            else:
+                label_intervals = np.load(
+                    self.root_folder / f"{label}.npy", allow_pickle=True
+                )
+
+            for start, end in label_intervals:
+                mask = (valid_times >= start) & (valid_times < end)
+                out[mask, l] = True
+
+        return out, valid
```

Comment:
could we remove returning `valid` from here please? 
This is in #73 and @schewskone is working on it for other interpolators in #81